tests/kafkatest/tests/core/transactions_test.py

```diff
@@ -0,0 +1,207 @@
+                           err_msg="Failed to see timely deregistration of \
+                           hard-killed broker %s" % str(node.account))
+                self.kafka.start_node(node)
+
+    def create_and_start_message_copier(self, input_partition, transactional_id):
+        message_copier = TransactionalMessageCopier(
+            context=self.test_context,
+            num_nodes=1,
+            kafka=self.kafka,
+            transactional_id=transactional_id,
+            consumer_group=self.consumer_group,
+            input_topic=self.input_topic,
+            input_partition=input_partition,
+            output_topic=self.output_topic,
+            max_messages=-1,
+            transaction_size=self.transaction_size
+        )
+        message_copier.start()
+        wait_until(lambda: message_copier.alive(message_copier.nodes[0]),
+                   timeout_sec=10,
+                   err_msg="Message copier failed to start after 10 s")
+        return message_copier
+
+    def bounce_copiers(self, copiers, clean_shutdown):
+        for _ in range(3):
+            for copier in copiers:
+                wait_until(lambda: copier.progress_percent() >= 20.0,
+                           timeout_sec=30,
+                           err_msg="%s : Message copier didn't make enough progress in 30s. Current progress: %s" \
+                           % (copier.transactional_id, str(copier.progress_percent())))
+                self.logger.info("%s - progress: %s" % (copier.transactional_id,
+                                                        str(copier.progress_percent())))
+                copier.restart(clean_shutdown)
+
+    def create_and_start_copiers(self):
+        copiers = []
+        copiers.append(self.create_and_start_message_copier(
+            input_partition=0,
+            transactional_id=self.first_transactional_id
+        ))
+        copiers.append(self.create_and_start_message_copier(
+            input_partition=1,
+            transactional_id=self.second_transactional_id
+        ))
+        return copiers
+
+    def copy_messages_transactionally(self, failure_mode, bounce_target):
+        copiers = self.create_and_start_copiers()
+        clean_shutdown = False
+        if failure_mode == "clean_bounce":
+            clean_shutdown = True
+
+        if bounce_target == "brokers":
+            self.bounce_brokers(clean_shutdown)
+        elif bounce_target == "clients":
+            self.bounce_copiers(copiers, clean_shutdown)
+
+        for copier in copiers:
+            wait_until(lambda: copier.is_done,
+                       timeout_sec=30,
+                       err_msg="%s - Failed to copy all messages in  %ds." %\
+                       (copier.transactional_id, 30))
+        self.logger.info("finished copying messages")
+
+    @cluster(num_nodes=8)
+    @matrix(failure_mode=["clean_bounce", "hard_bounce"],
+            bounce_target=["brokers", "clients"])
```

Comment:
I don't think so: with the bug fix for KAFKA-5351, they seem to be quite stable.